File: frontend/src/types/chapter.ts
export interface ChapterSummary {
  id: number;
  title: string;
  chapterNumber: number;
  viewCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface ChapterDetail {
  storyId: number;
  storyTitle: string;
  storySlug: string;
  chapterId: number;
  chapterTitle: string;
  chapterNumber: number;
  content: string;
  previousChapterNumber: number | null;
  nextChapterNumber: number | null;
}

File: backend/story-service/src/main/java/com/example/storyservice/chapter/Chapter.java
package com.example.storyservice.chapter;

import com.example.storyservice.story.Story;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "chapters", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"story_id", "chapter_number"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Chapter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "story_id", nullable = false)
    private Story story;

    @Column(nullable = false)
    private String title;

    @Column(name = "chapter_number", nullable = false)
    private Integer chapterNumber;

    @Column(nullable = false, columnDefinition = "LONGTEXT")
    private String content;

    @Column(name = "view_count")
    @Builder.Default
    private Long viewCount = 0L;

    @Column(name = "source_url")
    private String sourceUrl;

    @Column(nullable = false)
    @Builder.Default
    private Boolean imported = false;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
